src/commands.ts

```diff
@@ -57,5 +60,31 @@ export namespace Commands {
 
     export const JAVA_RESOLVEPATH = "java.resolvePath";
 
+    export const JAVA_PROJECT_GETMAINMETHOD = "java.project.getMainMethod";
+
+    export const JAVA_PROJECT_EXPORTJAR = "java.project.exportJar";
+
     export const VSCODE_OPEN_FOLDER = "vscode.openFolder";
 }
+
+export const JAVA_BUILD_WORKSPACE = "java.workspace.compile";
+
+export const JAVA_EXECUTE_WORKSPACE_COMMAND = "java.execute.workspaceCommand";
+
+export const JAVA_RESOLVE_BUILD_FILES = "vscode.java.resolveBuildFiles";
+
+export function executeJavaLanguageServerCommand(...rest) {
+    return executeJavaExtensionCommand(JAVA_EXECUTE_WORKSPACE_COMMAND, ...rest);
+}
+
+export async function executeJavaExtensionCommand(commandName: string, ...rest) {
+    // TODO: need to handle error and trace telemetry
+    const javaExtension = getJavaExtension();
+    if (!javaExtension) {
+        throw new JavaExtensionNotEnabledError(`Cannot execute command ${commandName}, VS Code Java Extension is not enabled.`);
```

Comment:
this extension is hard depend on vs code java. Which means the vscode java will always be activated when this extension starts. So this check is not necessary for now.